internal/services/datafactory/data_factory_trigger_schedules_data_source.go

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package datafactory
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/datafactory/2018-06-01/factories"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+type TriggerSchedulesDataSource struct{}
+
+type TriggerSchedulesDataSourceModel struct {
+	DataFactoryID string   `tfschema:"data_factory_id"`
+	Items         []string `tfschema:"items"`
+}
+
+func (d TriggerSchedulesDataSource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"data_factory_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: factories.ValidateFactoryID,
+		},
+	}
+}
+
+func (d TriggerSchedulesDataSource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"items": {
+			Type:     pluginsdk.TypeList,
+			Computed: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+	}
+}
+
+func (d TriggerSchedulesDataSource) ModelObject() interface{} {
+	return &TriggerSchedulesDataSourceModel{}
+}
+
+func (d TriggerSchedulesDataSource) ResourceType() string {
+	return "azurerm_data_factory_trigger_schedules"
+}
+
+func (d TriggerSchedulesDataSource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model TriggerSchedulesDataSourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return err
+			}
+
+			client := metadata.Client.DataFactory.TriggersClient
+
+			dataFactoryId, err := factories.ParseFactoryID(model.DataFactoryID)
+			if err != nil {
+				return err
+			}
+
+			iter, err := client.ListByFactoryComplete(ctx, dataFactoryId.ResourceGroupName, dataFactoryId.FactoryName)
+			if err != nil {
+				if v, ok := err.(autorest.DetailedError); ok {
+					if utils.ResponseWasNotFound(autorest.Response{Response: v.Response}) {
+						return fmt.Errorf("fetching trigger list from Azure Data Factory %s", dataFactoryId.ID())
+					}
+				} else {
+					return fmt.Errorf("fetching triggers list from Azure Data Factory %q: %+v", dataFactoryId.ID(), err)
+				}
+				return fmt.Errorf("fetching triggers list from Azure Data Factory %q: %+v", dataFactoryId.ID(), err)
+			}
```

Comment:
I think we can simplify these
```suggestion
						return fmt.Errorf("fetching triggers list for %s", dataFactoryId)
					}
				} else {
					return fmt.Errorf("fetching triggers list for %s: %+v", dataFactoryId, err)
				}
				return fmt.Errorf("fetching triggers list for %s: %+v", dataFactoryId, err)
			}
```